packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/albums/AddAssetsToAlbum.kt

```diff
@@ -24,11 +19,11 @@ internal class AddAssetsToAlbum(
   private val albumId: String,
   copyToAlbum: Boolean,
   private val promise: Promise
-) : AsyncTask<Void?, Void?, Void?>() {
```

Comment:
One very important thing here. You can't just remove async tasks and not replace them with some other forms of async solutions like coroutines. Your current solution will block the native module queue because all the disk operations will be dispatched to the same place where all calls from js are processed. This can lead to a lot of lags in your application, especially on low-end devices. 

Could you replace the `AsyncTask` with coroutines? To do it, you have to create a coroutine scope in the module and then dispatch all the work there.  [Here](https://github.com/expo/expo/pull/14029/files) you can find a small example.

Reply:
No problem, I was wondering about this too. I wasn't sure whether to use the Coroutine builder on the AsyncFunction or create a scope myself.